src/ruby/lib/grpc/generic/rpc_server.rb

```diff
@@ -354,6 +354,51 @@ def run
 
     alias_method :run_till_terminated, :run
 
+    # runs the server with signal handlers
+    def run_till_terminated_or_interrupted(signals)
+      valid_signals = Signal.list
+
+      stop_server = false
+      stop_server_cv = ConditionVariable.new
+      stop_server_mu = Mutex.new
+      stop_server_thread = Thread.new do
+        loop do
+          break if stop_server
+          stop_server_mu.synchronize { stop_server_cv.wait(stop_server_mu, 60) }
```

Comment:
please make this wait interval parameterized

Reply:
Parameterized to run_till_terminated_or_interrupted. 
Added an optional parameter for user to specify the timeout.